src/main/java/logic/Storage.java

```diff
@@ -0,0 +1,107 @@
+package logic;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import exception.InvalidSaveFileException;
+import tasks.Deadlines;
+import tasks.Events;
+import tasks.Task;
+import tasks.ToDos;
+
+/**
+ * Handles saving of the list and reading it upon startup.
+ */
+public class Storage {
+    private String fileLocation;
+    public Storage(String fileLocation) {
+        this.fileLocation = fileLocation;
+    }
+
+    /**
+     * Called upon startup. Reads save file if there is one, else creates
+     * a blank save file.
+     *
+     * @return A List populated by the tasks.Task from previous save file.
+     * @throws InvalidSaveFileException If there is an issue with creating the save file.
+     */
+    public List<Task> readFile() throws InvalidSaveFileException {
+        List<Task> toDoList = new ArrayList<Task>();
+        File dir = new File("data");
+        if (!dir.exists()) {
+            dir.mkdir();
+        }
+        try {
+            File file = new File(fileLocation);
+            if (file.exists()) {
+                Scanner s = new Scanner(file);
+                while (s.hasNext()) {
+                    String entry = s.nextLine();
+                    DateTimeFormatter dtf2 = DateTimeFormatter.ofPattern("HH:mm MMM d yyyy");
+                    if (entry.startsWith("[T]")) {
+                        ToDos t = new ToDos(entry.substring(7));
+                        if (entry.contains("✓")) {
+                            t.completeTask();
+                        }
+                        toDoList.add(t);
+                    } else if (entry.startsWith("[D]")) {
+                        int index = entry.indexOf("(");
+                        String datetime = entry.substring(index + 5, entry.length() - 1);
+                        LocalDateTime ldt = LocalDateTime.parse(datetime, dtf2);
+                        Deadlines d = new Deadlines(entry.substring(7, index - 1),
+                                ldt);
+                        if (entry.contains("✓")) {
+                            d.completeTask();
+                        }
+                        toDoList.add(d);
+                    } else {
```

Comment:
Would it be better to explicitly check `else if (entry.startsWith("[E]"))` here instead of just `else`? Then, in the `else` block, perhaps it may be better to throw an exception since we don't expect the progam to reach that state.